### PR TITLE
Add hero search 

### LINKED ISSUE #247: Create a huge variation of search
screen shot 2015-09-04 at 11 46 56

### PR DESCRIPTION
**Pull request by kdzwinel**
Closes #247

To recreate search from http://brainly.com/ I had to add two new modifiers to `mint-input`: mint-input--large and mint-input--subtle-border: 
 
screen shot 2015-09-07 at 16 01 27

"Hero search" looks like this on desktop:
screen shot 2015-09-07 at 16 01 50

And like this on mobile:
screen shot 2015-09-07 at 16 02 01

### CODE REVIEW

File: src/components/form-elements/_form-elements.scss [hunk 1]
```diff
@@ -123,6 +128,7 @@ $include-html: false !default;
     height: $inputHeight;
     line-height: $inputHeight;
     appearance: none;
+    transition: border 300ms;
```
Comment (kdzwinel) [inline src/components/form-elements/_form-elements.scss]: unrelated change, border color smoothly transitions between colors now - it's subtle but looks much nicer than a sudden color change

File: src/components/form-elements/_form-elements.scss [hunk 2]
```diff
@@ -144,13 +150,13 @@ $include-html: false !default;
     &--valid {
       border-color: $inputValidBorderColor;
       &:focus {
-        border-color: $inputValidBorderColor;
+        border-color: $inputFocusValidBorderColor;
```
Comment (kdzwinel) [inline src/components/form-elements/_form-elements.scss]: Unrelated change:  color of the border for 'valid' and 'invalid' fields changes on focus (it wasn't changing before)

File: src/components/search/_search.scss [hunk 3]
```diff
@@ -5,6 +5,7 @@ $include-html: false !default;
 
   .mint-search {
     @include component;
+    overflow: visible;
```
Comment (kdzwinel) [inline src/components/search/_search.scss]: `component` sets `overflow: hidden` we don't want that for search because it cuts the shadow of the `mint-input--subtle-border`

File: src/components/form-elements/_form-elements.scss (hunk 1 above)
Comment (mr-mig) [inline src/components/form-elements/_form-elements.scss]: great idea

**Comment by mr-mig** [comment]
:+1:

File: src/components/form-elements/_form-elements.scss (hunk 2 above)
Comment (vergilius) [inline src/components/form-elements/_form-elements.scss]: :+1:

**Comment by vergilius** [comment]
:+1:

File: src/components/form-elements/_form-elements.scss (hunk 1 above)
Comment (aju) [inline src/components/form-elements/_form-elements.scss]: nice:)

**Comment by aju** [comment]
:+1:
